ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -144,6 +146,10 @@ public void execute(
 
     final DataSource dataSource = getDataSource(config, metaStore, insertValues);
 
+    if (!verifyInsertValuesAllowed(insertValues.getColumns(), dataSource)) {
+      throw new KsqlException("Cannot insert into a HEADER column");
```

Comment:
It'd be good to display the column header name. Perhaps throwing the exception from inside the method would be helpful so you can customize the message. What about renaming the method too `validateInsert()` or `throwIfInsertOnHeadersColumns()`?

Reply:
Updated the messages